File: obj-c/28_usov_ch_37/TestProject/TestProject/ViewController2.m
//
//  ViewController2.m
//  TestProject
//
//  Created by Виталий Цыганенко on 08.04.2023.
//

#import "ViewController2.h"

@interface ViewController2 ()

@property (strong, nonatomic) IBOutlet UITextField* alertTitleTextField;
@property (strong, nonatomic) IBOutlet UITextField* alertMessageTextField;

@end

@implementation ViewController2

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

-(IBAction) onUnwindAction: (UIStoryboardSegue*) unwindSegue {
    NSLog(@"ViewController2 unwind action");
}

-(IBAction) showAlert {
    
    UIAlertController* alert = [UIAlertController
        alertControllerWithTitle: self.alertTitleTextField.text.length == 0 ? @"Default title" : self.alertTitleTextField.text
        message: self.alertMessageTextField.text.length == 0 ? @"Default message" : self.alertMessageTextField.text
        preferredStyle: UIAlertControllerStyleAlert];
    
    void (^handler1) (UIAlertAction* action) =
        ^(UIAlertAction* action) {
            NSLog(@"alert: button OK is chosen");
    };
    
    void (^handler2) (UIAlertAction* action) =
        ^(UIAlertAction* action) {
            NSLog(@"alert: button %@ is chosen", [action title]);
    };
    
    UIAlertAction* buttonOK = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:handler1];
    UIAlertAction* buttonCancel = [UIAlertAction actionWithTitle:@"Cancel" style:UIAlertActionStyleCancel handler:handler2];
    UIAlertAction* buttonDel = [UIAlertAction actionWithTitle:@"Del" style:UIAlertActionStyleDestructive
                                                      handler: ^(UIAlertAction* action) {
        NSLog(@"alert: button %@ is chosen", [action title]);
    }];
    
    [alert addAction:buttonOK];
    [alert addAction:buttonCancel];
    [alert addAction:buttonDel];
    
    [self presentViewController:alert animated:YES completion:^(void){NSLog(@"alert is shown");}];
}

- (IBAction)onTapGesture:(id)sender {
    [self.alertTitleTextField resignFirstResponder];
    [self.alertMessageTextField resignFirstResponder];
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: swift/01_usov_ch_37/TestProject/TestProject/ViewController2.swift
//
//  ViewController2.swift
//  TestProject
//
//  Created by Виталий Цыганенко on 10.04.2023.
//

import UIKit

class ViewController2: UIViewController {

    @IBOutlet var alertTitleTextField: UITextField!
    @IBOutlet var alertMessageTextField: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    @IBAction func onUnwindAction(unwindSegue: UIStoryboardSegue) {
        print("ViewController2::onUnwindAction")
    }
    
    @IBAction func showAlert() {
        let titleAvailable = (alertTitleTextField.text != nil && alertTitleTextField.text!.isEmpty == false)
        let messageAvailable = (alertMessageTextField.text != nil && alertMessageTextField.text!.isEmpty == false)
        
        let alert = UIAlertController(
            title: titleAvailable ? alertTitleTextField.text : "Default title",
            message: messageAvailable ? alertMessageTextField.text : "Default message",
            preferredStyle: .alert)
        
        let handler1 = { (_ action:UIAlertAction) -> Void in
            print("alert: button OK is chosen")
        }
        
        let handler2 = { (_ action:UIAlertAction) -> Void in
            print("alert: button", action.title!, "is chosen")
        }
        
        let buttonOK = UIAlertAction(title:"OK", style: .default, handler: handler1)
        let buttonCancel = UIAlertAction(title:"Cancel", style: .cancel, handler: handler2)
        let buttonDel = UIAlertAction(title:"Del", style: UIAlertAction.Style.destructive,
                                      handler: { (_ action:UIAlertAction) -> Void in
            print("alert: button", action.title!, "is chosen")
        })
        
        alert.addAction(buttonOK)
        alert.addAction(buttonCancel)
        alert.addAction(buttonDel)
        
        self.present(alert, animated: true, completion:
                        { () -> Void in
            print("alert is shown")
        })
    }
    
    @IBAction func onTapGesture() {
        alertTitleTextField.resignFirstResponder()
        alertMessageTextField.resignFirstResponder()
    }

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destination.
        // Pass the selected object to the new view controller.
    }
    */

}
